paimon-core/src/main/java/org/apache/paimon/operation/FileStoreCommitImpl.java

```diff
@@ -1274,4 +1308,101 @@ static ConflictCheck noConflictCheck() {
     public static ConflictCheck mustConflictCheck() {
         return latestSnapshot -> true;
     }
+
+    boolean doCommit(
+            long newSnapshotId,
+            Path newSnapshotPath,
+            Snapshot newSnapshot,
+            boolean useSerializableIsolation,
+            boolean useLock) {
+        try {
+            boolean committed = tryAtomicCommitNewSnapshot(newSnapshotPath, newSnapshot);
+            if (committed) {
+                if (useSerializableIsolation && !useLock) {
+                    // If the useSerializableIsolation is turned on,And we not use lock service,
+                    // we will remove all HINTs,as it may provide incorrect information.
+                    snapshotManager.removeSnapshotEarliestHint();
```

Comment:
If Serializable is turned on, and the locking service is not used, we should delete all HINT files, as they are always unreliable.